### PR TITLE
Fix run_e2e_workflow

### LINKED ISSUE #30: Prow jobs marked as failure even though Gubernator indicates job succeeded
Example:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/kubeflow_tf-operator/388/kubeflow-tf-operator-presubmit/35/

Job is marked as success and no test failures. There is no error in the log.

The issue is that the pod is exiting with non zero exit code. If we look at the pod logs (via prow jobs UI) we see the following exceptoin

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py", line 294, in <module>
    success = main()
  File "/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py", line 284, in main
    return run(args, file_handler)
  File "/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py", line 184, in run
    if results["status"]["phase"] != "Succeeded":
TypeError: list indices must be integers, not str
```

This exception happens after the logs have been uploaded to GCS which is why it doesn't show up in the gubernator logs.

/cc @lluunn

### PR DESCRIPTION
**Pull request by jlewi**
* run_e2e_workflow.py is exiting with an exception causing all prow jobs to be marked as a failure

* Add an E2E test to kubeflow/testing. This provides a way to test run_e2e_workflow.py as part of the presubmits.

Fix #30

### CODE REVIEW

File: py/kubeflow/testing/argo_client.py [hunk 1]
```diff
@@ -65,11 +63,12 @@ def wait_for_workflows(client, namespace, names,
...
     if datetime.datetime.now() + polling_interval > end_time:
       raise util.TimeoutError(
-        "Timeout waiting for workflow {0} in namespace {1} to finish.".format(
-          name, namespace))
+        "Timeout waiting for workflows {0} in namespace {1} to finish.".format(
+          ",".join(names), namespace))
 
     time.sleep(polling_interval.seconds)
 
+  return []
```
Comment (Jimexist) [inline py/kubeflow/testing/argo_client.py]: new line after this pls

File: py/kubeflow/testing/py_checks.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+  # different results.
+  util.run(["pylint", "--version"])
+
+  # kubeflow_testing is imported as a submodule so we should exclude it
+  # TODO(jlewi): We should make this an argument.
+  dir_excludes = ["dashboard/frontend/node_modules", "kubeflow_testing",
+                  "vendor",]
+  full_dir_excludes = [os.path.join(os.path.abspath(args.src_dir), f) for f in
+                       dir_excludes]
+  includes = ["*.py"]
```
Comment (Jimexist) [inline py/kubeflow/testing/py_checks.py]: would really benefit from Python3's Pathlib

File: py/kubeflow/testing/py_checks.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+  failed_files = []
+  rc_file = os.path.join(args.src_dir, ".pylintrc")
+  for root, dirs, files in os.walk(os.path.abspath(args.src_dir),
+                                   topdown=True):
+    # excludes can be done with fnmatch.filter and complementary set,
+    # but it's more annoying to read.
+    exclude = False
```
Comment (Jimexist) [inline py/kubeflow/testing/py_checks.py]: can be extracted into a smaller functions:

```python
def should_exclude(root, full_dir_excludes):
  for e in full_dir_excludes:
    if root.startswith(e):
      return True
  else:
    return False
```

File: py/kubeflow/testing/py_checks.py (hunk 2 above)
Comment (jlewi) [reply py/kubeflow/testing/py_checks.py]: Ack.

**Comment by jlewi** [comment]
Woo Hoo passing run
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/kubeflow_testing/33/kubeflow-testing-presubmit/24/

**Pull request merged**